### ISSUE TITLE
dune-project not formatted with @fmt

### ISSUE THREAD

**Issue by bbc2** [ocaml/dune#3223]
When I run `dune build @fmt`, I would expect `dune-project` to be checked as well as other `dune` files.  Any reason for it not to be checked?  I understand that the first line of that file (e.g. `(lang dune 2.3)`) is a bit of an exception, but the rest could probably benefit from auto formatting, no?

## Specifications

- Version of `dune` (output of `dune --version`): 2.3.1
- Version of `ocaml` (output of `ocamlc --version`): 4.09.0
- Operating system (distribution and version): Arch Linux

**Comment by diml** [comment]
Indeed. Reading the code in `src/dune/format_rules.ml`, it just seems like an omission. More precisely, we recognise `dune` files but not `dune-project` files (in the `match Path.Source.basename file with | "dune" ...`).

/cc @emillon

**Comment by diml** [comment]
Do you want to submit a PR?

**Comment by emillon** [comment]
That's in part intentional, because it does not look to great at the moment (#2291). As explained in that bug, the generic s-expression formatter does not know about any syntax (stanza, records, etc). 

For dune files it's difficult to have that without adding introspection capabilities to the language (there's no explicit AST), but for `dune-project` we might be able to do better.

**Comment by diml** [comment]
That makes sense. Since we have a good idea how to add introspection to the language, we should wait for that so that we have a general mechanism.

**Comment by jberdine** [comment]
It would make sense to treat dune-workspace files at the same time once the formatting of dune-project files is pretty enough to enable.
